homeassistant/components/esphome/voice_assistant.py

```diff
@@ -124,57 +142,112 @@ async def _iterate_packets(self) -> AsyncIterable[bytes]:
         while data := await self.queue.get():
             yield data
 
+    def _event_callback(self, event: PipelineEvent) -> None:
+        """Handle pipeline events."""
+
+        try:
+            event_type = _VOICE_ASSISTANT_EVENT_TYPES.from_hass(event.type)
+        except KeyError:
+            _LOGGER.warning("Received unknown pipeline event type: %s", event.type)
+            return
+
+        data_to_send = None
+        if event_type == VoiceAssistantEventType.VOICE_ASSISTANT_STT_END:
```

Comment:
This code could benefit from using an event handler design, where each event has a corresponding handler function. We look up the correct function using the event type. We have this util to help with designs like that:

https://github.com/home-assistant/core/blob/7d2243344e1aaa05ac8986bf9c9047693d35c60f/homeassistant/util/decorator.py#L11-L22